### PR TITLE
Typescript support

### PR DESCRIPTION
**Pull request by sim51**
* Adding typescript support
* Adding prettier

### CODE REVIEW

File: examples/basic.ts [hunk 1]
```diff
@@ -3,23 +3,23 @@ import erdosRenyi from 'graphology-generators/random/erdos-renyi';
 import randomLayout from 'graphology-layout/random';
 import chroma from 'chroma-js';
 import faker from 'faker';
-import WebGLRenderer from '../src/renderers/webgl';
+import WebGLRenderer from '../src/renderers/webgl/index';
```
Comment (Yomguithereal) [inline examples/basic.ts:6]: TypeScript ne gère pas la résolution des index? C'est une convention?

Comment (sim51) [reply examples/basic.ts:6]: Without it, it doesn't compile. 
But perhaps we can change this behaviour in the TS configuration (`moduleResolution` ?)

Comment (sim51) [reply examples/basic.ts:6]: Related SO question : https://stackoverflow.com/questions/44028806/should-the-index-ts-be-resolved-by-typescript-as-default-module-file

**Comment by sim51** [comment]
We also need to discuss about the linter (for now it's unchanged ...)
Usually I use this config :

```
{
  "parser": "@typescript-eslint/parser",
  "plugins": ["@typescript-eslint"],
  "extends": ["plugin:@typescript-eslint/recommended"],
  "rules": {
    "@typescript-eslint/camelcase": "off"
  }
}
```

What do you think ?

```

File: examples/basic.ts (hunk 1 above)
Comment (Yomguithereal) [reply examples/basic.ts:6]: Does this mean TS people consuming the lib need to point `sigma/renderers/webgl/index`?

**Comment by Yomguithereal** [comment]
@sim51 I am going to merge so I can advance things related to the mouse captor without losing your work. I will open a separate issue on what we will need to decide regarding the linter/prettier & the build etc.

File: examples/basic.ts (hunk 1 above)
Comment (sim51) [reply examples/basic.ts:6]: I will say it depends of their TS configuration.
I have a react TS project where it's not needed to put the index. So I think we can find a TS config that works for the omission of 'index'.

**Pull request merged**